link.go

```diff
@@ -49,9 +49,14 @@ type link struct {
 	doneErr  error         // contains the mux error state; ONLY written to by the mux and MUST only be read from after done is closed!
 	closeErr error         // contains the error state returned from closeLink(); ONLY closeLink() reads/writes this!
 
-	session    *Session                // parent session
-	source     *frames.Source          // used for Receiver links
-	target     *frames.Target          // used for Sender links
+	session *Session       // parent session
+	source  *frames.Source // used for Receiver links
+
+	// target is of type:
+	// - *[frames.Target] - for typical Senders
+	// - *[frames.CoordinatorTarget] - for TransactionControllers
+	target any // used for Sender links
```

Comment:
Can we not use a generic type parameter with a constraint on `frames.Target | frames.CoordinatorTarget`?

Reply:
I considered this but felt like it would have to propagate quite a bit. For instance, the link is a field within Sender so I'd need to also make Sender generic.

I could combat that by splitting out the guts from Sender, having some internal impl, etc...

I kind of still want to do it, but I figured it'd be a bit big and overshadow the work.

Reply:
Yeah probably more unnecessary churn.